From 0b638ece826c816a2387834975c9b5854dc94f6b Mon Sep 17 00:00:00 2001
From: Igor Opaniuk <igor.opaniuk@foundries.io>
Date: Fri, 21 Jul 2023 18:08:03 +0200
Subject: [PATCH] k3: set env variable device_type

Set environment variable "device_type", which stores device
sec state: gp, hs-fs, hs-se.

Upstream-Status: Inappropriate [lmp specific]
Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>
---
 arch/arm/mach-k3/common.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-k3/common.c b/arch/arm/mach-k3/common.c
index c04f898a55..14619fabdb 100644
--- a/arch/arm/mach-k3/common.c
+++ b/arch/arm/mach-k3/common.c
@@ -600,6 +600,8 @@ void spl_board_prepare_for_linux(void)
 
 int misc_init_r(void)
 {
+	int device_type;
+
 	if (IS_ENABLED(CONFIG_TI_AM65_CPSW_NUSS)) {
 		struct udevice *dev;
 		int ret;
@@ -611,12 +613,20 @@ int misc_init_r(void)
 			printf("Failed to probe am65_cpsw_nuss driver\n");
 	}
 
+	device_type = get_device_type();
 	/* Default FIT boot on HS-SE devices */
-	if (get_device_type() == K3_DEVICE_TYPE_HS_SE) {
+	if (device_type == K3_DEVICE_TYPE_HS_SE) {
 		env_set("boot_fit", "1");
 		env_set("secure_rprocs", "1");
+		env_set("device_type", "hs-se");
 	} else {
+		if (device_type == K3_DEVICE_TYPE_HS_FS) {
+			env_set("device_type", "hs-fs");
+		} else {
+			env_set("device_type", "gp");
+		}
 		env_set("secure_rprocs", "0");
+
 	}
 
 	return 0;
-- 
2.34.1

